language/stdlib/modules/DesignatedDealer.move

```diff
@@ -225,12 +280,34 @@ module DesignatedDealer {
                 dealer.window_inflow == old(dealer.window_inflow) + amount);
         ensures tier_index < len(old(dealer).tiers);
         ensures dealer.window_inflow <= old(dealer).tiers[tier_index];
+
+        include TieredMint<CoinType>;
+    }
+
+    spec schema TieredMint<CoinType> {
+        tc_account: signer;
+        dd_addr: address;
+        amount: u64;
+        tier_index: u64;
+        include Roles::AbortsIfNotTreasuryCompliance{account: tc_account};
+        aborts_if amount == 0 with Errors::INVALID_ARGUMENT;
+        include AbortsIfDealerNotExistAt;
+        aborts_if !exists<TierInfo<CoinType>>(dd_addr);
+        let tier_info = global<TierInfo<CoinType>>(dd_addr);
+        aborts_if tier_info.window_start + ONE_DAY > max_u64();
+        aborts_if LibraTimestamp::spec_now_microseconds() <= tier_info.window_start + ONE_DAY && tier_info.window_inflow + amount > max_u64();
```

Comment:
You can also do the following:

```
include LibraTimestamp::spec_now_microseconds() <= tier_info.window_start + ONE_DAY 
       ==> TieredMintAbortsIfToday;
...
schema TieredMintAbortsIfToday {
   aborts_if tier_info.window_inflow + amount > MAX_U64;
   aborts_if tier_info.window_inflow + amount > tier_info.tiers[tier_index;
   aborts_if amount > tier_info.tiers[tier_index];
}
```

Or use a let to avoid repeating the premise, though the above is suited for real complex conditional aborts or ensures.

Reply:
I have cleaned this up and also rewritten the code to eliminate all overflows.